### PR TITLE
Added lambda attn to hyperparameter tuning

### PR DESCRIPTION
**Pull request by JoeCx**
# Overview

**Type of Change:** New Feature

**Summary:** Added lambda attention to the hyperparameter tuning process in the grad cam training program. Tuning this should increase the accuracy because lambda attention directly controls how much weight you give the CAM-guided loss relative to the classification loss (total_loss = pred_loss + self.lambda_attn * attn_loss).

## UI/UX Implementation

No changes made.

## Model Updates

This section describes changes that were made to the models used by your application.

### Data Models

No changes made.

### Object-Oriented Models

No changes made.

## End-to-End Testing Instructions

Run the cam_training_simulator.py file and select hyperparameter tuning. Once an optimal lambda attention value has been found change the self.lambda_attn variable in the constructor of the training program to the correct value and rerun training.

### CODE REVIEW

**Review by tmckeown311 — APPROVED** [review]
Changes worked in testing. Not much to say for review other than good job
